src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java

```diff
@@ -0,0 +1,390 @@
+        lx.addEffect(scrambleEffect);
+        lx.addEffect(staticEffect);
+        lx.addEffect(rotationEffect);
+        lx.addEffect(spinEffect);
+        lx.addEffect(speedEffect);
+        lx.addEffect(colorStrobeTextureEffect);
+        lx.addEffect(fadeTextureEffect);
+        lx.addEffect(acidTripTextureEffect);
+        lx.addEffect(candyTextureEffect);
+        lx.addEffect(candyCloudTextureEffect);
+
+        registerEffectControlParameter(speedEffect.speed, "3707000050abae", 1, 0.4);
+        registerEffectControlParameter(speedEffect.speed, "3707000050a916", 1, 5);
+        registerEffectControlParameter(colorEffect.rainbow, "3707000050a98f");
+        registerEffectControlParameter(colorEffect.mono, "3707000050aafe");
+        registerEffectControlParameter(colorEffect.desaturation, "3707000050a969");
+        registerEffectControlParameter(colorEffect.sharp, "3707000050aafc");
+        registerEffectControlParameter(blurEffect.level, "3707000050a973", 0.65);
+        registerEffectControlParameter(spinEffect.spin, "3707000050ab2c", 0.65);
+        registerEffectControlParameter(ghostEffect.amount, "3707000050aaf2", 0, 0.16, 1);
+        registerEffectControlParameter(scrambleEffect.amount, "3707000050a8cc", 0, 1, 1);
+        registerEffectControlParameter(colorStrobeTextureEffect.amount, "3707000050a946", 0, 1, 1);
+        registerEffectControlParameter(fadeTextureEffect.amount, "3707000050a967", 0, 1, 1);
+        registerEffectControlParameter(acidTripTextureEffect.amount, "3707000050a953", 0, 1, 1);
+        registerEffectControlParameter(candyCloudTextureEffect.amount, "3707000050a92d", 0, 1, 1);
+        registerEffectControlParameter(staticEffect.amount, "3707000050a8b3", 0, .3, 1);
+        registerEffectControlParameter(candyTextureEffect.amount, "3707000050aafc", 0, 1, 5);
+
+        effectKnobParameters =
+                new LXListenableNormalizedParameter[] {
+                    colorEffect.hueShift,
+                    colorEffect.mono,
+                    colorEffect.desaturation,
+                    colorEffect.sharp,
+                    blurEffect.level,
+                    speedEffect.speed,
+                    spinEffect.spin,
+                    candyCloudTextureEffect.amount
+                };
+    }
+
+    public void registerPattern(EntwinedBasePattern pattern, String nfcSerialNumber) {
+        registerPattern(pattern, nfcSerialNumber, 2);
+    }
+
+    public void registerPattern(
+            EntwinedTriggerablePattern pattern, String nfcSerialNumber, int apc40DrumpadRow) {
+        registerVisual(pattern, nfcSerialNumber, apc40DrumpadRow, VisualType.Pattern);
+    }
+
+    void registerVisual(
+            EntwinedTriggerablePattern pattern,
+            String nfcSerialNumber,
+            int apc40DrumpadRow,
+            VisualType visualType) {
+
+        LXBlend t = new DissolveBlend(lx);
+
+        // NOTE(meawoppl) @Slee same question below.  re `.setDuration(dissolveTime);`
```

Comment:
There are no longer per-pattern transition settings. The transition type and duration are now simply set at the channel level.